### PR TITLE
feat(similarity-embedding): Add endpoint to call backfill script

### PR DESCRIPTION
**Pull request by jangjodi**
Create endpoint to call the backfill script [here](https://github.com/getsentry/sentry/pull/68466)

### CODE REVIEW

File: src/sentry/api/endpoints/project_similar_issues_embeddings_records.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.tasks.backfill_seer_grouping_records import backfill_seer_grouping_records
+
+region_silo_endpoint
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:11]:
```suggestion
```

File: src/sentry/api/endpoints/project_similar_issues_embeddings_records.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class ProjectSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
+    owner = ApiOwner.ISSUES
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+
+    def post(self, request: Request, project) -> Response:
+        if not features.has("projects:similarity-embeddings-grouping", project):
+            return Response(status=404)
+        backfill_seer_grouping_records.delay(project)
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:23]: could we make it so only superusers can call this endpoint? (can just use `is_active_superuser` for now).  e.g. https://github.com/getsentry/sentry/blob/90d35a1104422d3d40974032aeb673e8f31a77c7/src/sentry/api/bases/sentryapps.py#L492

File: src/sentry/api/endpoints/project_similar_issues_embeddings_records.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.tasks.backfill_seer_grouping_records import backfill_seer_grouping_records
+
+
+
+@region_silo_endpoint
+class ProjectSimilarIssuesEmbeddingsRecords(ProjectEndpoint):
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:13]: what do you think about adding `Backfill` somewhere in this endpoint name? (and in the url path in urls.py?) unless we want to add other functionality to this endpoint

Comment (jangjodi) [reply src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:13]: yeah that's a good idea - I'll make this change

File: src/sentry/api/endpoints/project_similar_issues_embeddings_records.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from rest_framework.request import Request
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:1]: should this file be removed from the PR?

File: src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.auth.superuser import is_active_superuser
+from sentry.tasks.backfill_seer_grouping_records import backfill_seer_grouping_records
+
+region_silo_endpoint
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:12]:
```suggestion
```

Comment (jangjodi) [reply src/sentry/api/endpoints/project_backfill_similar_issues_embeddings_records.py:12]: I'm not sure why this keeps re-appearing - it doesn't show up in my code editor locally either

**Review by JoshFerge — APPROVED** [review]

File: src/sentry/api/endpoints/project_similar_issues_embeddings_records.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from rest_framework.response import Response
```
Comment (jangjodi) [inline src/sentry/api/endpoints/project_similar_issues_embeddings_records.py:1]: TODO: delete this file

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**